### ISSUE TITLE
Streaming Telemed images to 3D Slicer : No frame received by the device 

### ISSUE THREAD

**Issue by glennmullerar** [PlusToolkit/PlusLib#944]
Hi everyone, 

I'm new using plusServer. I want to stream images from my Telemed ArtUs to 3D Slicer through plusServer.

When, I launch  the server, this error appears (see screenshot).

I Use :  

- PlusServer Telemed-Win32 2.8.0
- Slicer 4.11
- Echo Wave II : 4.1.1

![plusserver_error](https://user-images.githubusercontent.com/75022384/166938543-91113b23-5067-4639-b8e8-9d939eb21acb.PNG)

**Comment by adamrankin** [comment]
Could you upload the log as a text file?

**Comment by glennmullerar** [comment]
Here is the log.

Thank's


[051022_093710_PlusLog.txt](https://github.com/PlusToolkit/PlusLib/files/8659387/051022_093710_PlusLog.txt)

**Comment by adamrankin** [comment]
Hmm, no errors in the log with regards to connection. @markasselin Does telemed have its own software you can test the device with?

@glennmullerar are you able/interested in making a debug build to try and diagnose?

**Comment by adamrankin** [comment]
@glennmullerar Can you confirm the device works with the Telemed software?

**Comment by glennmullerar** [comment]
Yes it’s work perfectly. I can save dicom files too.

**Comment by lassoan** [comment]
There are slight differences between Telemed driver/firmware versions and if it does not match the application software version (EchoWave, Plus, ...) then that can cause this issue.

@glennmullerar What EchoWave software have you tested with?
@ungi What EchoWave software version do we use?

**Comment by ungi** [comment]
That's right, I remember we needed to update the driver for a newer Telemed probe. I use EchoWave II 3.6.2. My driver package installer says ver. 1.17.2.
We plan to transition to 4.x Telemed software versions, which are also available in 64-bit. So we could use Telemed with a 64-bit build of PLUS. But I don't think that has happened yet.

**Comment by glennmullerar** [comment]
@lassoan I tested with Echwave 4.1.1. I had to test with 3.6.2 version of echowave ?

**Comment by lassoan** [comment]
To use Plus you would need to downgrade to 3.6.2 driver and potentially also the firmware; or we would need to create a Plus package that is built with Telemed 4.1.1 SDK.

At some point we need to switch to the latest SDK version anyway (if all our hardware supports it) for improved compatibility, robustness, and potentially new features. @Sunderlandkyl could you create an experimental build with the latest Telemed SDK so that @glennmullerar can test it?

**Comment by glennmullerar** [comment]
The streaming works with Echo Wave 3.6.2, but now I see a white bande with Echo Wave II 3.6.2 (See screenshot). Maybe I need the right firmware and/or driver for this Echo Wave version ?


![echowave_3_6_2](https://user-images.githubusercontent.com/75022384/168098974-2a3c58d9-ea40-4458-ae62-398445c7d216.PNG)

**Comment by MarionCrouzier** [comment]
Hi everybody,

I do have the same problem.

I am trying to install again 3D slicer on another computer in my lab. I use:
PlusServer Telemed-Win - 2.8.0.20191105 -Telemed-Win32
Echo Wave II : 4.1.1
(of note, on other computers I have Echo Wave II : 4.1.0 and it works fine).

At the moment I do have this same error when launching Plus (No frame received by the device), more specifically:
"|ERROR|119.379000|SERVER> TelemedUltrasound::SetDepthMm failed: not connected to hardware interface| in :\D\PSNPTeb\PlusLib\src\PlusDataCollection\Telemed\TelemedUltrasound.cxx(955)

|ERROR|119.400000|SERVER> vtkPlusTelemedVideoSource parameter setting failed: DepthMm=100| in :\D\PSNPTeb\PlusLib\src\PlusDataCollection\Telemed\vtkPlusTelemedVideoSource.cxx(273)

|ERROR|119.400000|SERVER> No frame received by the device| in :\D\PSNPTeb\PlusLib\src\PlusDataCollection\Telemed\vtkPlusTelemedVideoSource.cxx(144)"

The US is looking good in Echowave though (even if I do have to run it as an administrator to have it work).

What was the conclusion of the above discussion, and what do you recommend I do to solve my problem?

Thank you for your help,

Kind regards,

Marion

**Comment by ungi** [comment]
Hi, we had this problem before, and Telemed recommended that we run PLUS in compatibility mode. You can set that up in Windows by right-clicking on the green PLUS launcher icon, select Properties, then Compatibility, check "Run this program in compatibility mode for: Windows 8". After that, I started receiving images with content. Let us know if this solves your problem please.

**Comment by AurelieS** [comment]
Thank you so much for your reactivity, it works !! :D
Best regards,
Aurélie
